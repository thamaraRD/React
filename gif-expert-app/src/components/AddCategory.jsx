import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewValue }) => {
  const [inputValue, setInputValue] = useState('');

  //DesestructuraciÃ³n del evento
  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (e) => {
    e.preventDefault();

    const newInputValue = inputValue.trim();
    // eslint-disable-next-line no-unused-expressions
    newInputValue.length <= 0 ? null : onNewValue(newInputValue )
    setInputValue('');
  }

  return (
    <form aria-label='form' onSubmit={ onSubmit }>
      <input
        type="text"
        placeholder="Busca tu Gif..."
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.PropTypes = {
  onNewValue: PropTypes.func.isRequired
}