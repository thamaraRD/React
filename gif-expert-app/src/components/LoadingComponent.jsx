
export const LoadingComponent = () => {
  return (
    <div className="loader"></div>
  )
};
