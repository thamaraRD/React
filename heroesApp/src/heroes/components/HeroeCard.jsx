
export const HeroeCard = ({
  id,
  superhero,
  publisher,
  alter_ego,
  first_appearance,
  characters,
}) => {

  

  return (

<div className="col">
  <div className="card">
    <div className="row no-gutters">
    <div className="col-4">
      <img src="" alt={ superhero } className="card-img" />
    </div>
    </div>
  </div>
</div>
  
)

};
