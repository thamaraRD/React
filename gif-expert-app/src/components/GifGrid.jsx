import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { LoadingComponent } from "./LoadingComponent";

export const GifGrid = ({ category }) => {
  const { images, loader } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loader && <LoadingComponent />}
      <div className="card-grid">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};
