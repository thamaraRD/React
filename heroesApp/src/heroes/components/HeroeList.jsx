import { getHeroesByPublisher } from "../helpers/getHeroesByPublisher";
import { HeroeCard } from '../components/HeroeCard'
import { useMemo } from "react";

export const HeroeList = ({ publisher }) => {
//No es necesario en este caso el useMemo
  const heroes =  useMemo( () => getHeroesByPublisher(publisher), [publisher] );

  return (
     
      <div className="row rows-cols-1 row-cols-md-3 g-3">
        { heroes.map( heroe => (
        
          <HeroeCard key={heroe.id} 
            { ...heroe }
          />

        ))
      }
      </div>
  
  )
}
