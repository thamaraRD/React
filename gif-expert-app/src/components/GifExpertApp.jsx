import PropTypes from 'prop-types';
import { useState } from "react";
import { AddCategory } from "./AddCategory";
import { GifGrid } from "./GifGrid";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['Rugrats']);

  const handleAddCategory = (onNewValue) => {
    if (
      !categories.some(
        (category) => category.toLowerCase() === onNewValue.toLowerCase()
      )
    ) {
      setCategories([onNewValue, ...categories]);
    }
  };

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory onNewValue={handleAddCategory} />

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};

GifExpertApp.PropTypes = {
  title: PropTypes.string.isRequired
};
