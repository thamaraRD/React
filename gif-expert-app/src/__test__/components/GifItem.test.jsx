import { screen, render } from "@testing-library/react";
import { GifItem } from "../../components/GifItem";

describe('test on <GifItem />', () => { 

    const title = 'Rugrats';
    const url = 'https://rugrats.com/carlitos.jpg'

    test('should match to Snapshot', () => { 

        const { container } = render ( <GifItem title={ title } url={ url } /> );
        expect ( container ).toMatchSnapshot();
    });

    test('should show the URL and the indicated ALT', () => { 
        render ( <GifItem title={ title } url={ url } /> );

        // expect (screen.getByRole('img').src).toBe(url);
        // expect (screen.getByRole('img').alt).toBe(title);

        const { src, alt } = screen.getByRole('img');
        expect (src).toBe(url);
        expect (alt).toBe(title);
        
    });

    test('should show {title} in the component', () => { 

        render ( <GifItem title={ title } url={ url } /> );
        expect ( screen.getByText(title) ).toBeTruthy();
    });

});