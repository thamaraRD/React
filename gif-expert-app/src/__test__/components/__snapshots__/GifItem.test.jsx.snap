// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <GifItem /> should match to Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Rugrats"
      src="https://rugrats.com/carlitos.jpg"
    />
    <p>
      Rugrats
    </p>
  </div>
</div>
`;
